core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -160,6 +161,14 @@ public GenericDataFile(org.apache.avro.Schema avroSchema) {
     this.keyMetadata = ByteBuffers.toByteArray(keyMetadata);
   }
 
+  GenericDataFile(String filePath, FileFormat format, PartitionData partition,
+                  long fileSizeInBytes, Metrics metrics,
+                  ByteBuffer keyMetadata, List<Long> splitOffsets, Integer fileType) {
+    this(filePath, format, partition, fileSizeInBytes, metrics, splitOffsets);
```

Comment:
this can be  ..
```
this(filePath, format, partition, fileSizeInBytes, metrics,  keyMetadata, splitOffsets);
this.fileType = fileType;
```

.. for better reuse of code.

Reply:
OK, no problem!